developer-docs-site/docs/nodes/staking.md

```diff
@@ -0,0 +1,236 @@
+:::
+
+## Epoch
+
+An epoch in the Aptos blockchain is defined as a duration of time, in seconds, during which a number of blocks are voted on by the validators, the validator set is updated, and the rewards are distributed to the validators. 
+
+:::tip
+Currently an epoch on the Aptos blockchain is defined as sixty seconds (one hour).
+:::
+
+### Triggers at the epoch start
+
+:::tip
+See [https://github.com/aptos-labs/aptos-core/blob/0daade4f734d1ba29a896b00d7ddde2249e87970/aptos-move/framework/aptos-framework/sources/configs/stake.move#L862](https://github.com/aptos-labs/aptos-core/blob/0daade4f734d1ba29a896b00d7ddde2249e87970/aptos-move/framework/aptos-framework/sources/configs/stake.move#L862) for the full code.
+:::
+
+At the start of each epoch, the following key events are triggered:
+
+- Update the validator set by adding the pending active validators to the active validators set and by removing the pending inactive validators from the active validators set.
+- The status of the stake is updated. For example any pending active stake is updated to active stake and any pending inactive stake is updated to inactive stake.
+- The voting power of the validator in the validator set is updated.
+- Rewards are distributed to the active validators (i.e., validators in the validator set).
+- Rewards are distributed to the validators who requested to leave but have not yet been removed.
+
+## Rewards
+
+Rewards for staking are calculated by using `rewards_rate`, an annual percentage yield (APY), using the below two numbers:
+
+- Your current total staked amount.
+- Your remaining lock up time.
+
+Rewards accrue as a compound interest on your current staked amount. 
+
+:::tip Set by the governance
+
+The `rewards_rate` is set by the Aptos governance.
+
+:::
+
+### Rewards paid every epoch
+
+Rewards are paid every epoch. Any reward you earned at the end of current epoch is added to your staked amount. The reward at the end of the next epoch is calculated based on your increased staked amount (i.e., original staked amount plus the added reward), and so on.
+
+### Rewards formula
+
+The lock up period has both the minimum required time and the maximum allowed time. If you lock up for the maximum period, then you will receive the maximum `rewards_rate`, i.e., a measure of APY. amount of the possible rewards. For example, if the APY is 10%, then if you lock up for the full maximum period, then you will receive the maximum reward, calculated at 10% APY on the staked amount in your account.
+
+Shown below the formula used to calculate your rewards:
+
+```
+Reward = Maximum possible reward * (Remaining lockup / Maximum lockup) * (Number of successful votes / Total number of blocks in the current epoch)
+```
```

Comment:
it feels we should equate based upon total transactions too -- maybe a very small multiplier.